complex/ft_config.c

```diff
@@ -228,38 +228,76 @@ static struct key_t keys[] = {
 		.val_size = sizeof(((struct ft_set *)0)->op) / FI_ATOMIC_OP_LAST,
 	},
 	{
+		.str = "datatype",
+		.offset = offsetof(struct ft_set, datatype),
+		.val_type = VAL_NUM,
+		.val_size = sizeof(((struct ft_set *)0)->datatype) / FI_DATATYPE_LAST,
+	},
+	{
 		.str = "mode",
 		.offset = offsetof(struct ft_set, mode),
 		.val_type = VAL_NUM,
 		.val_size = sizeof(((struct ft_set *)0)->mode) / FT_MAX_PROV_MODES,
 	},
-	{
+	{	//keep this for backwards compatibility
```

Comment:
We don't need to maintain backwards compatibility for the tests.  The ofiwg agreed that we don't want to deal with that maintenance cost.  So if caps were replaced with test_class, we can just move forward and update the config files.  Any config files in the field will just need to be updated.